### PR TITLE
mgr/dashboard: Fix flaky QA tests

### PR DESCRIPTION
**Pull request by p-na**
Fixes some errors in QA test runs by passing all errors as message in the response. I likely should have done that from the beginning.

See http://qa-proxy.ceph.com/teuthology/sage-2018-09-10_02:11:51-rados-master-distro-basic-smithi/2999041/teuthology.log

Signed-off-by: Patrick Nawracay <pnawracay@suse.com>

### CODE REVIEW

File: qa/tasks/mgr/dashboard/test_osd.py [hunk 1]
```diff
@@ -99,13 +99,15 @@ def test_create_lost_destroy_remove(self):
         self.assertStatus(200)
 
     def test_safe_to_destroy(self):
-        self._get('/api/osd/5/safe_to_destroy')
+        self._get('/api/osd/50/safe_to_destroy')
```
[the remark this answers is not in the document]
Comment (p-na) [reply qa/tasks/mgr/dashboard/test_osd.py]: Never. The usual behaviour of `safe-to-destroy` is to return `yes, safe to destroy` for any OSD that doesn't exist. That's also the reason why I increased the number; to be on the safe side.

[the remark this answers is not in the document]
Comment (rjfd) [reply qa/tasks/mgr/dashboard/test_osd.py]: hmm... in that case wouldn't be safer to get the list of all available OSDs and then choose a value that does not exist in that list?

[the remark this answers is not in the document]
Comment (p-na) [reply qa/tasks/mgr/dashboard/test_osd.py]: Yeah, it would be safer. I'll look into it.

**Comment by tchaikov** [comment]
```
2018-09-12T07:28:16.870 INFO:tasks.cephfs_test_runner:Starting test: test_scrub (tasks.mgr.dashboard.test_osd.OsdTest)
2018-09-12T07:28:16.871 INFO:tasks.mgr.dashboard.helper:request POST to https://mira038.front.sepia.ceph.com:7789/api/osd/0/scrub?deep=False
2018-09-12T07:28:16.878 INFO:tasks.ceph.mgr.x.mira038.stderr:2018-09-12 07:28:16.873 7fc8fd7fd700 -1 mgr.server reply reply (11) Resource temporarily unavailable osd.0 is not up
2018-09-12T07:28:16.886 INFO:tasks.ceph.mgr.x.mira038.stdout:::ffff:172.21.0.51 - - [12/Sep/2018:07:28:16] "POST /api/osd/0/scrub?deep=False HTTP/1.1" 400 88 "" "python-requests/2.12.5"
2018-09-12T07:28:16.888 INFO:tasks.cephfs_test_runner:test_scrub (tasks.mgr.dashboard.test_osd.OsdTest) ... FAIL
2018-09-12T07:28:16.888 INFO:tasks.mgr.dashboard.helper:request POST to https://mira038.front.sepia.ceph.com:7789/api/osd/0/mark_in
2018-09-12T07:28:16.901 INFO:tasks.cephfs_test_runner:
2018-09-12T07:28:16.901 INFO:tasks.cephfs_test_runner:======================================================================
2018-09-12T07:28:16.902 INFO:tasks.cephfs_test_runner:FAIL: test_scrub (tasks.mgr.dashboard.test_osd.OsdTest)
2018-09-12T07:28:16.902 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2018-09-12T07:28:16.902 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2018-09-12T07:28:16.902 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2018-09-12-1250/qa/tasks/mgr/dashboard/test_osd.py", line 52, in test_scrub
2018-09-12T07:28:16.902 INFO:tasks.cephfs_test_runner:    self.assertStatus(200)
2018-09-12T07:28:16.902 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2018-09-12-1250/qa/tasks/mgr/dashboard/helper.py", line 325, in assertStatus
2018-09-12T07:28:16.902 INFO:tasks.cephfs_test_runner:    self.assertEqual(self._resp.status_code, status)
2018-09-12T07:28:16.902 INFO:tasks.cephfs_test_runner:AssertionError: 400 != 200
```
http://pulpito.ceph.com/kchai-2018-09-12_06:58:08-rados-wip-kefu-testing-2018-09-12-1250-distro-basic-mira/

**Comment by callithea** [comment]
Unfortunately one Job failed: http://pulpito.ceph.com/laura-2018-09-12_11:03:02-rados:mgr-wip-lpaduano-testing-24024-2-distro-basic-smithi/

**Comment by tchaikov** [comment]
and http://pulpito.ceph.com/kchai-2018-09-12_16:09:03-rados-wip-kefu-testing-2018-09-12-1250-distro-basic-mira/  some tests failed with ```Test failure: test_scrub (tasks.mgr.dashboard.test_osd.OsdTest)```

**Comment by callithea** [comment]
QA run was successful: http://pulpito.ceph.com/laura-2018-09-13_10:57:06-rados:mgr-wip-lpaduano-testing-24024-3-distro-basic-smithi/ and http://pulpito.ceph.com/laura-2018-09-13_12:27:44-rados:mgr-wip-lpaduano-testing-24024-3-distro-basic-smithi/

**Pull request merged**